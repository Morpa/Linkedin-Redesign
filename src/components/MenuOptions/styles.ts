import styled, { css, DefaultTheme } from 'styled-components'

const linkModifiers = {
  default: (theme: DefaultTheme) => css`
    color: ${theme.colors.black};
  `,
  active: (theme: DefaultTheme) => css`
    color: ${theme.colors.blue};

    &:after {
      content: '';
      width: 8rem;
      height: 0.1rem;
      background: ${theme.colors.blue};
      position: absolute;
      left: 50%;
      transform: translateX(-50%);
      bottom: 0;
    }
  `
}

type linkProps = {
  isActive?: boolean
}

export const Option = styled.a<linkProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    flex-direction: column;
    position: relative;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    line-height: ${theme.spacings.small};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small}
      ${theme.spacings.xxsmall} ${theme.spacings.small};

    strong {
      margin-top: ${theme.spacings.xxsmall};
      font-weight: ${theme.font.bold};
    }

    > svg {
      width: 2.4rem;
    }

    &:hover {
      color: ${theme.colors.blue};
      cursor: pointer;
    }

    ${!isActive && linkModifiers.default(theme)};
    ${isActive && linkModifiers.active(theme)};
  `}
`
