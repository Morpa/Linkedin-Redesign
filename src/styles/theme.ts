export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans- serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  },

  colors: {
    primary: 'linear-gradient(180deg, #0077B5 0%, #0E6795 100%)',
    blue: '#0275B1',
    black: '#181818',
    white: '#FFFFFF',
    bg: '#F7F9FB',
    green: '#02B033',
    gray: '#CECECE',
    grayDark: '#747474',
    orange: 'linear-gradient(215.49deg, #FFB75E 5.4%, #ED8F03 93.45%)'
  },

  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  }
} as const
