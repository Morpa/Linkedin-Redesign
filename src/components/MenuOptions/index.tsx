import { useRouter } from 'next/router'
import Link from 'next/link'

import * as S from './styles'

export type MenuOptionsProps = {
  link: string
  linkName: string
  icon?: React.ReactNode
  activeLink?: string
}

const MenuOptions = ({
  link,
  linkName,
  icon,
  activeLink
}: MenuOptionsProps) => {
  const router = useRouter()

  return (
    <Link href={link} passHref prefetch={false}>
      <S.Option isActive={activeLink === router.pathname}>
        {!!icon && icon}
        <strong>{linkName}</strong>
      </S.Option>
    </Link>
  )
}

export default MenuOptions
