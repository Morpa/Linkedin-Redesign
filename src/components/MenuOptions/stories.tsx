import { Story, Meta } from '@storybook/react/types-6-0'
import { RssFeed } from '@styled-icons/material-outlined'

import MenuOptions, { MenuOptionsProps } from '.'

export default {
  title: 'MenuOptions',
  component: MenuOptions,
  args: {
    link: '/link',
    linkName: 'My link',
    icon: <RssFeed />,
    activeClassName: 'active'
  },
  argTypes: {
    icon: {
      type: ''
    }
  }
} as Meta

export const Default: Story<MenuOptionsProps> = (args) => (
  <MenuOptions {...args} />
)
