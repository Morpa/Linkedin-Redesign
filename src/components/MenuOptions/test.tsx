import { screen } from '@testing-library/react'
import { RssFeed } from '@styled-icons/material-outlined'

import MenuOptions from '.'
import { renderWithTheme } from 'utils/tests/helpers'
import theme from 'styles/theme'

const props = {
  link: '/link',
  linkName: 'My link',
  icon: <RssFeed />
}

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({ pathname: '/link' }))
}))

describe('<MenuOptions />', () => {
  it('should render options correctly', () => {
    const { container } = renderWithTheme(<MenuOptions {...props} />)

    expect(screen.getByText(/my link/i)).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /my link/i })).toHaveAttribute(
      'href',
      '/link'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the icon when it is passed', () => {
    renderWithTheme(
      <MenuOptions {...props} icon={<RssFeed data-testid="icon" />} />
    )

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render  when  is active', () => {
    renderWithTheme(<MenuOptions {...props} activeLink="/link" />)

    expect(screen.getByRole('link', { name: /my link/i })).toHaveStyle({
      color: theme.colors.blue
    })
  })
})
