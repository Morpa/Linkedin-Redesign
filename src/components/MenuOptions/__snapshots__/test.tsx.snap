// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuOptions /> should render options correctly 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  line-height: 2.4rem;
  padding: 1.6rem 2.4rem 0.8rem 2.4rem;
  color: #181818;
}

.c0 strong {
  margin-top: 0.8rem;
  font-weight: 600;
}

.c0 > svg {
  width: 2.4rem;
}

.c0:hover {
  color: #0275B1;
  cursor: pointer;
}

<a
  class="c0"
  href="/link"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <circle
      cx="6.18"
      cy="17.82"
      r="2.18"
    />
    <path
      d="M4 4.44v2.83c7.03 0 12.73 5.7 12.73 12.73h2.83c0-8.59-6.97-15.56-15.56-15.56zm0 5.66v2.83c3.9 0 7.07 3.17 7.07 7.07h2.83c0-5.47-4.43-9.9-9.9-9.9z"
    />
  </svg>
  <strong>
    My link
  </strong>
</a>
`
